src/drivers/roboclaw/RoboclawSerialDevice.hpp

```diff
@@ -0,0 +1,66 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file RoboclawSerialDevice.hpp
+ * @brief
+ * @author Matthias Grob <maetugr@gmail.com>
+ */
+
+#include "RoboclawDriver/RoboclawDriver.hpp"
+#include <px4_platform_common/px4_work_queue/ScheduledWorkItem.hpp>
+
+class RoboclawDevice : public RoboclawWritableInterface
+{
+public:
+	RoboclawDevice(const char *port);
+	~RoboclawDevice();
+	int init();
+	void printStatus();
+
+private:
+	void Run();
+	size_t writeCallback(const uint8_t *buffer, const uint16_t length) override;
+	int setBaudrate(const unsigned baudrate);
+
+	static constexpr size_t READ_BUFFER_SIZE{256};
+	static constexpr int DISARM_VALUE = 0;
+	static constexpr int MIN_THROTTLE = 100;
+	static constexpr int MAX_THROTTLE = 1000;
+
+	char _port[20] {};
+	int _serial_fd{-1};
+	VescDriver _vesc_driver;
```

Comment:
VescDriver?

Reply:
This entire file was a draft of me refactoring the driver. Needs to be merged with the rest :eyes: